### ISSUE TITLE
429 handling

### ISSUE THREAD

**Issue by confact** [guilhermesad/rspotify#54]
Would be nice with some type of inbuilt handling of Spotify's rate limit flow. 

They set a limit and when it is reached they send a 429 with a retry after header which includes seconds when you can retry after.

Is it anyway to fix this?

**Comment by guilhermesad** [comment]
@confact that's an interesting idea. Maybe also including a message to the user telling it'll retry after X seconds. I'm open to pull requests on that :)

**Comment by MentalPower** [comment]
This has the potential to be a backwards incompatible change. Make sure it's optional or at least can be turned off. I have some retry logic around this already that I implemented outside of rspotify. My reasoning is that I have some use cases where I want to fail fast instead of blocking and retrying.



—

**Comment by guilhermesad** [comment]
Yes, blocking could be unwanted, if this is to be implemented it has to be an optional feature.

**Comment by dgilperez** [comment]
Did someone got a solution for this eventually?

**Comment by rentziass** [comment]
Actually I did it in a fork, it's quite simple. May I do a pull request?

**Comment by rentziass** [comment]
It's just ages behind, I'll try to merge :smile:

**Comment by rentziass** [comment]
Sorry my fork is really to old and there are super crappy commits, BTW [here](https://github.com/rentziass/rspotify/blob/2e3d6654fadfccf6107d327aa075110bb3f1bfba/lib/rspotify/connection.rb#L68-L76) is how I implemented this.

For the option sake, I don't know how you may want to procede @guilhermesad, maybe a variable at module level and then as for `raw_response`?

**Comment by rentziass** [comment]
Ok, I'll quit the spam. I deleted my fork and started a new one. I'll let you know

**Comment by monfresh** [comment]
Another thing that would be nice, and which would be easier to implement, is to display the value of the retry header, so at least we know how long we need to wait.
